src/components/shared/VirtualizedScrollableGrid/index.js

```diff
@@ -0,0 +1,222 @@
+      const [rowIndex, columnIndex] = this.getGridIndexesFromAbsoluteIndex(
+        index,
+        colCount
+      );
+
+      gridElement.scrollToItem({
+        align: 'center',
+        rowIndex,
+        columnIndex,
+      });
+    }
+  }
+
+  updateResponsiveFeatures = debounce(() => {
+    const newState = Object.assign(
+      {},
+      this.setColumnCount(this.props),
+      this.setWidth(this.props)
+    );
+
+    if (this.componentMounted) {
+      this.setState(newState);
+    }
+  }, 300);
+
+  setColumnCount(props) {
+    const { theme, columnCount } = props;
+
+    let columnCountToSet = MIN_COLUMN_COUNT;
+
+    if (typeof columnCount === 'number') {
+      columnCountToSet = columnCount;
+    } else {
+      const currentColumnCount = this.findCurrentColumnCount(
+        columnCount,
+        theme
+      );
+      const maximumColumnCount = Math.max(...Object.values(columnCount));
+
+      if (currentColumnCount === null) {
+        columnCountToSet = maximumColumnCount;
+      } else {
+        columnCountToSet = currentColumnCount;
+      }
+    }
+
+    return {
+      columnCount: columnCountToSet,
+    };
+  }
+
+  setWidth(props) {
+    const { adaptWidthToElement, width } = props;
+    let widthToSet = 0;
+
+    if (adaptWidthToElement === null) {
+      widthToSet = width;
+    } else {
+      widthToSet = adaptWidthToElement.clientWidth;
+    }
+
+    return {
+      width: widthToSet,
+    };
+  }
+
+  handleScroll = scrollDestination => {
+    const { current } = this.gridRef;
+
+    if (current !== null) {
+      current.scrollTo(scrollDestination);
+    }
+  };
+
+  findCurrentColumnCount(columnCountObj, theme) {
+    const windowWidth = window.innerWidth;
+
+    const currentColumn = Object.keys(columnCountObj).find(breakpoint => {
+      const currentBreakpointResolution = theme.breakpoints[breakpoint];
+
+      return (
+        currentBreakpointResolution !== undefined &&
+        windowWidth < currentBreakpointResolution
+      );
+    });
+
+    if (currentColumn === undefined) return null;
+
+    return columnCountObj[currentColumn];
+  }
+
+  getAbsoluteIndexFromGridIndexes(rowIndex, columnIndex, colCount) {
+    return (rowIndex + 1) * colCount - (columnIndex + 1);
+  }
+
+  /**
+   * Returns the grid indexes by knowing the item index
+   * @param {Number} index Item Index
+   * @param {Number} colCount Column Count
+   * @returns {[Number, Number]} [RowIndex, ColIndex]
+   */
+  getGridIndexesFromAbsoluteIndex(index, colCount) {
+    return [Math.floor(index / colCount), index % colCount];
+  }
+
+  render() {
+    const { data } = this.props;
+    const { columnCount, width } = this.state;
+    const columnWidth = width / columnCount;
+    const rowCount = Math.ceil(data.length / columnCount);
+
+    // Weird fix, together with setting `List`'s direction to 'rtl', to render elements in the correct order
+    const newStyle = Object.assign({}, this.props.style, { direction: 'ltr' });
```

Comment:
Does it make sense to set this object each time during the render call?

Reply:
Kind of, because with each re-render, there might be some new `style` props coming, so I'm appending some new stuff to it.

Also, since `Object.assign` is pretty quick, memoizing this might actually be slower than it is right now.